### PR TITLE
Migrate AWS wazuh-qa integration tests to wazuh core repository

### LINKED ISSUE #17788: Refactor and migrate AWS QA Integration Tests to Core repository
## Description 

AWS integration tests found in the QA repository will be moved to wazuh/wazuh repository.

We need to ensure the migration is done correctly and tests are properly running.

First, the current wazuh-qa testing framework needs to be migrated and refactored to adapt to the new changes.

Then the AWS ITs can be migrated to wazuh core repository. 

Also, tests should be checked for a possible update on fixtures, use cases, features, etc.
  
## Related issues

- [x] https://github.com/wazuh/qa-integration-framework/issues/36.
- [x] https://github.com/wazuh/wazuh/issues/19053.
- [x] https://github.com/wazuh/wazuh/issues/19052.
- [x] https://github.com/wazuh/wazuh-qa/issues/4714.
- [x] https://github.com/wazuh/wazuh/issues/22514
- [x] https://github.com/wazuh/wazuh/issues/20975.
- [x] https://github.com/wazuh/qa-integration-framework/issues/133
- [x] https://github.com/wazuh/qa-integration-framework/issues/150

## Branches

- Epic branch: [17788-aws-integration-test-migration-and-refactor](https://github.com/wazuh/wazuh/tree/17788-aws-integration-test-migration-and-refactor)
- [qa-integration-framework](https://github.com/wazuh/qa-integration-framework/tree/17788-migrate-aws-integration-tests).
- [wazuh](https://github.com/wazuh/wazuh/tree/17788-aws-its-migration).

### LINKED ISSUE #19053: Migrate current wazuh-qa AWS ITs to wazuh core repository
| Affected integration |
|---|
| AWS|

## Description

As part of epic https://github.com/wazuh/wazuh/issues/17788, current wazuh-qa AWS ITs need to be migrated to core repository. This migration was already in progress and can be seen [starting here](https://github.com/wazuh/wazuh/issues/17788#issuecomment-1644544959) but due to the change in scope and more specifically to changes requested in [ framework pull request  ](https://github.com/wazuh/qa-integration-framework/pull/25) it was decided to be worked separately to better document the process of the migration and refactor.

This issue's scope is to only migrate the current test without making major changes. Minor changes such as cleaning code and better documentation (README) should be done.

## Checks

The only objective should be to run tests successfully without making major changes. Therefore, the tasks are:
 
- [x] Run all tests successfully utilizing the qa-integration-framework.
- [x] Clean code.
- [x] Update README.

### PR DESCRIPTION
**Pull request by EduLeon12**
|Related issue|
|---|
|Closes #19053 |

## Description

As part of #17788. This PR tends to migrate AWS QA Integration tests to wazuh core repository without making any major changes in the AWS test module.

It is expected that current changes will break current integration test pipelines making them fail until https://github.com/wazuh/qa-integration-framework/issues/36 is completed and merged.

A refactor to further improve the current test module and include all necessary changes to be able to run the pipeline successfully will be developed in #19052.

## Configuration options

README has been updated with new configuration options

## Logs/Alerts example

Log history can be found [here](https://github.com/wazuh/wazuh/issues/17788#issuecomment-1674971694). 

## Tests

Run of the full test suite: 

```
root@wazuh-master:/wazuh/tests/integration# pytest -x test_aws/ --disable-warnings
==================================== test session starts ====================================
platform linux -- Python 3.10.12, pytest-7.1.2, pluggy-1.2.0
rootdir: /wazuh/tests/integration, configfile: pytest.ini
plugins: testinfra-5.0.0, metadata-3.0.0, html-3.1.1
collected 195 items

test_aws/test_basic.py ................                                               [  8%]
test_aws/test_discard_regex.py ..............                                         [ 15%]
test_aws/test_log_groups.py ..                                                        [ 16%]
test_aws/test_only_logs_after.py .............................................x.      [ 40%]
test_aws/test_parser.py ..........................                                    [ 53%]
test_aws/test_path.py ..........................................                      [ 75%]
test_aws/test_path_suffix.py .........                                                [ 80%]
test_aws/test_regions.py ........................                                     [ 92%]
test_aws/test_remove_from_bucket.py ...sss.........                                   [100%]

============ 191 passed, 3 skipped, 1 xfailed, 7 warnings in 3723.08s (1:02:03) =============
```

### CODE REVIEW

**Review by nico-stefani — CHANGES_REQUESTED** [review]

File: tests/integration/conftest.py [hunk 1]
```diff
@@ -133,28 +136,119 @@ def load_wazuh_basic_configuration():
...
+
+    It uses the variable `file_to_monitor` to determinate the file to monitor. Default `LOG_FILE_PATH`
+
+    Args:
+        request (fixture): Provide information on the executing test function.
+    """
+    if hasattr(request.module, 'file_to_monitor'):
+        file_to_monitor = getattr(request.module, 'file_to_monitor')
+    else:
+        file_to_monitor = WAZUH_LOG_PATH
```
[the remark this answers is not in the document]
Comment (EduLeon12) [reply tests/integration/conftest.py:241]: As explained in [this comment](https://github.com/wazuh/wazuh/issues/17788#issuecomment-1673636630), the scope of this issue is to migrate the tests without making major enhancement changes. However, this proposal should be taken into consideration for https://github.com/wazuh/wazuh/issues/18355.

File: tests/integration/test_aws/utils.py
Comment (nico-stefani) [inline tests/integration/test_aws/utils.py:1]: Please, remove this file if don't we use it.

**Review by nico-stefani — CHANGES_REQUESTED** [review]

File: tests/integration/test_aws/utils.py
Comment (EduLeon12) [reply tests/integration/test_aws/utils.py:1]: My mistake, It is meant to be used in next step, but got caught on this branch, 
Done.

**Review by nico-stefani — APPROVED** [review]
LGMT

**Review by Selutario — APPROVED** [review]
This will be better reviewed in the epic PR, once the refactor is carried out.

**Pull request merged**